lib/semantics/resolve-names.cc

```diff
@@ -2790,6 +2791,25 @@ Symbol &SubprogramVisitor::PushSubprogramScope(
   return *symbol;
 }
 
+void SubprogramVisitor::PushBlockDataScope(const parser::Name &name) {
+  if (auto *prev{FindSymbol(name)}) {
+    if (prev->attrs().test(Attr::EXTERNAL) && prev->has<ProcEntityDetails>()) {
+      if (prev->test(Symbol::Flag::Subroutine) ||
+          prev->test(Symbol::Flag::Function)) {
+        Say2(name, "BLOCK DATA '%s' has been called"_err_en_US, *prev,
+            "Previous call of '%s'"_en_US);
+      }
+      EraseSymbol(name);
+    }
+  }
+  if (name.source.empty()) {
+    // Don't let unnamed BLOCK DATA conflict with unnamed PROGRAM
+    PushScope(Scope::Kind::BlockData, nullptr);
```

Comment:
According to the standard:
There shall not be more than one unnamed block data program unit in a program.
But this test is working:
block data
  common /test/ t1
end block data

block data
  integer x
end block data

Reply:
This "shall" requirement in the standard is something that can't be checked until link time.

Reply:
Even if they are in the same file?

Reply:
`SayAlreadyDeclared`  emits an error message; it doesn't check anything.